modules/nf-httpfs/src/main/nextflow/file/http/LatchPath.groovy

```diff
@@ -174,6 +180,74 @@ class LatchPath extends XPath {
         }
     }
 
+    void download(Path local) {
+        log.info "Downloading Latch file ${path.toString()} to ${local.toString()}"
+
+        def url = getSignedURL()
+
+        HttpRetryClient client = new HttpRetryClient()
+        def request =  HttpRequest.newBuilder()
+            .uri(url.toURI())
+            .header("Range", "bytes=0-0")
+            .GET()
+            .build()
+
+        def response = client.send(request)
+        if (![200, 206].contains(response.statusCode()))
+            throw new Exception("Failed to get file size: ${response.statusCode()}")
+
+        def contentRange = response.headers().firstValue('Content-Range')
+
+        def byteRangePrefix = "bytes 0-0/"
+        if (contentRange.empty || !contentRange.get().startsWith(byteRangePrefix))
+            throw new Exception("Failed to get file size: Content-Range invalid ${contentRange.get()}")
+
+        long fileSize = Long.parseLong(contentRange.get().substring(byteRangePrefix.length()))
+        long numParts = (long) Math.ceil(fileSize / (double) downloadPartSize)
```

Comment:
i remember i ran into some weirdness with longs not working with `Math.ceil` on the older java versions in the workflow base image, psa